socketio/src/client/builder.rs

```diff
@@ -149,11 +149,11 @@ impl ClientBuilder {
     #[allow(unused_mut)]
```

Comment:
Same as below, mutex lock should take a `mut self`.

Reply:
not sure what you meant but mut is indeed absolutelly useless, so i removed it

Reply:
Perhaps arc and mutex (previously rwlock) are created pretty early. Maybe we could create arc and mutex inside on connect method instead, but builder whould work with pure `Option<Callback>`s?

Reply:
Mh... does this make a difference?

Reply:
Codegen, but nothing else i think.

Reply:
Then I would rather leave it as is.